### PR TITLE
Deprecate `--fix` flag of `calicoctl checksystem`

### LINKED ISSUE #539: The --fix flag to calicoctl checksystem should be depracated.
All it does it try to modprobe some modules, but felix will load them itself.

Remember to search for references to `checksystem --fix` in e.g. cloud-init files.

### PR DESCRIPTION
**Pull request by alexhersh**
+ Fix #539
+ Remove all references to the --fix flag in guides
+ Remove checksystem calls in startup scripts
+ Make module checks more reliable

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -78,7 +78,7 @@ calico_containers/calico-node.tar: caliconode.created
 	docker save --output calico_containers/calico-node.tar calico/node
 
 st: binary calico_containers/busybox.tar calico_containers/routereflector.tar calico_containers/calico-node.tar run-etcd
-	sudo dist/calicoctl checksystem --fix
+	sudo dist/calicoctl checksystem
```
Comment (tomdee) [inline Makefile]: I think we should just remove this call. It's not needed to run every time we run the tests. Developers system and CI systems will either work or not and it won't change. If it's not working, then they can run the command manually.

File: calico_containers/calico_ctl/checksystem.py [hunk 2]
```diff
@@ -93,73 +92,43 @@ def normalize_version(version):
...
-                                     (module, module)
-                system_ok = False
+            print >> sys.stderr, "WARNING: Unable to detect the %s " \
+                                 "module." % module
+            system_ok = False
     return system_ok
 
 
-def _check_ip_forwarding(fix):
+def _check_ip_forwarding():
```
Comment (tomdee) [inline calico_containers/calico_ctl/checksystem.py]: Hmm, might be better to rebase on your other PR and just remove this section...

**Comment by tomdee** [comment]
Let's get your other two PRs in, then rebase and I'll take another quick look.

**Comment by tomdee** [comment]
Can you rebase onto master.

**Comment by tomdee** [comment]
ha ha. And the tests are failing. I think it will be fixed by applying my code review feedback.

File: calico_containers/calico_ctl/checksystem.py [hunk 3]
```diff
@@ -44,22 +44,18 @@ def checksystem(arguments):
...
-    attempt to fix any issues it encounters. If any fixes fail, it will
-    exit(1). Fix will automatically be set to True if the user specifies --fix
-    at the command line.
+    Checks that the system is setup correctly.
 
-    :param fix: if True, try to fix any system dependency issues that are
-    detected.
     :param quit_if_error: if True, quit with error code 1 if any issues are
-    detected, or if any fixes are unsuccesful.
+    detected.
```
Comment (alexwlchan) [inline calico_containers/calico_ctl/checksystem.py]: The docstring doesn’t mention the libnetwork argument.

**Comment by tomdee** [comment]
LGTM - we will need to test the Ubuntu and CoreOS vagrant files but that can be done as part of the next release.

**Pull request merged**